File: P/495_Teemo_Attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int total = duration;
        
        for(int i = timeSeries.length - 2; i >= 0; i--)
            total += Math.min(duration, timeSeries[i + 1] - timeSeries[i]);
        
        return total;
    }
}

File: 주영/Any/495. Teemo Attacking.js
var findPoisonedDuration = function (timeSeries, duration) {
  let answer = duration;

  for (let i = 1; i < timeSeries.length; i++) {
    answer += Math.min(timeSeries[i] - timeSeries[i - 1], duration);
  }
  return answer;
};
